cli/src/main/java/org/jboss/as/cli/impl/aesh/cmd/deployment/AbstractUndeployCommand.java

```diff
@@ -118,10 +118,13 @@ static void undeployName(CommandContext ctx, String name,
         try {
             ModelNode request = buildRequest(ctx,
                     name, allServerGroups, serverGroups, keepContent, headers);
-            final ModelNode result = ctx.
-                    getModelControllerClient().execute(request);
-            if (!Util.isSuccess(result)) {
-                throw new CommandException(Util.getFailureDescription(result));
+
+            if (request.get(Util.STEPS).isDefined()) {
```

Comment:
This should be:

```
if (request.hasDefined(Util.STEPS)) {
```

A call to ModelNode.get is actually a mutating call, as it creates the child if it isn't present. You don't want to mutate just to check existence.

I'll leave it to @jfdenise to do the official review. I do think though that there should be some comment to explain why that check is relevant, i.e. why is it guaranteed that only with the presence of 'steps' that 'request' is useful. It seems a bit fragile, since in general an op without a 'steps' param is valid.

Reply:
Even if the request is not used anymore, there is no problem in changing this to hasDefined